clicommand/agent_start.go

```diff
@@ -835,6 +835,18 @@ var AgentStartCommand = cli.Command{
 			}
 		}
 
+		// Treat a negative signal grace period as relative to the cancel grace period
+		if cfg.SignalGracePeriodSeconds < 0 {
+			if cfg.SignalGracePeriodSeconds < cfg.CancelGracePeriod*-1 {
+				return fmt.Errorf(
+					"signal-grace-period-seconds (%d) must be greater than the negative cancel-grace-period (%d)",
+					cfg.SignalGracePeriodSeconds,
+					cfg.CancelGracePeriod*-1,
+				)
+			}
```

Comment:
Hmmm. Comparisons with negatives get a bit harder to understand. At least, I don't think it's natural to negate `CancelGracePeriod` when the already-negated quantity is `SignalGracePeriod`. Phrase in terms of positive values?

```go
			if cfg.CancelGracePeriod < -cfg.SignalGracePeriodSeconds {
				return fmt.Errorf(
					"cancel-grace-period (%d) must be at least as big as signal-grace-period-seconds (%d)",
					cfg.CancelGracePeriod,
					cfg.SignalGracePeriodSeconds,
				)
			}
```

...or...

```go
			if cfg.CancelGracePeriod + cfg.SignalGracePeriodSeconds < 0 {
				return fmt.Errorf(
					"cancel-grace-period (%d) + signal-grace-period-seconds (%d) must be non-negative",
					cfg.CancelGracePeriod,
					cfg.SignalGracePeriodSeconds,
				)
			}
```